Makefile

```diff
@@ -17,29 +18,32 @@ help:
 	@echo "Example configuration located at: $(DOCSDIR)"
 	@echo
 
-.PHONY: help Makefile
-
-install: remove_legacy
+.PHONY: root_check
+root_check:
 	@if ! [ "$(shell id -u)" = 0 ]; then echo "You must be root to perform this action."; exit 1; fi
-	@echo "Installing wireguard-initramfs ..."
+
+.PHONY: install_deps
+install_deps: root_check
 	@apt update && apt install initramfs-tools
+
+.PHONY: install_files
+install_files:
 	@mkdir -p "$(TARGETDIR)"
 	@touch "$(TARGETDIR)/private_key"
 	@chmod 0600 "$(TARGETDIR)/private_key"
 	@touch "$(TARGETDIR)/pre_shared_key"
 	@chmod 0600 "$(TARGETDIR)/pre_shared_key"
 	@cp -vn config "$(TARGETDIR)/config"
 	@chmod 0644 "$(TARGETDIR)/config"
-	@cp -v hooks "$(INITRAMFS)/hooks/wireguard"
-	@chmod 0755 hooks "$(INITRAMFS)/hooks/wireguard"
-	@cp -v init-premount "$(INITRAMFS)/scripts/init-premount/wireguard"
-	@chmod 0755 init-premount "$(INITRAMFS)/scripts/init-premount/wireguard"
-	@cp -v init-bottom "$(INITRAMFS)/scripts/init-bottom/wireguard"
-	@chmod 0755 init-bottom "$(INITRAMFS)/scripts/init-bottom/wireguard"
-	-@mkdir -p "$(DOCSDIR)/examples"
-	@chmod -R 0755 "$(DOCSDIR)"
-	@cp -v config "$(DOCSDIR)/examples/config"
-	@chmod 0644 "$(DOCSDIR)/examples/config"
+	@install -vD hooks "$(INITRAMFS)/hooks/wireguard"
```

Comment:
I would prefer explicitly specifying perms here in case of wonky consumer umasks.

Reply:
`install` is independent of the users umask and will always use 0755, if you do not specify any other modes.